extend glossary with terms for upcoming Web API and related components (#166)
OCR-D/ocrd-website: site/en/spec/glossary.md

### (OCR-D-)Backend

Software component of a server concerned with network operations; e.g. Python library with request handlers, implementing service discovery and network-capable workspace data management.

### (OCR-D-)Workflow-Runtime-Library

Model-specific (?) software component of a server or processor; e.g. Python API in [OCR-D/core](https://github.com/OCR-D/core) providing classes for all essential functional components (OcrdPage, OcrdMets, Workspace, Resolver, Processor, ProcessorTask, Workflow, WorkflowTask...), including mechanisms for signalling and orchestration of workflows, on top of which components (from processor to controller) can be realized.

### (OCR-D-)Workflow-Engine

Central software component of the controller, executing workflows, including control structures (linear/parallel/incremental).

### Processor

> A processor is a method provided by a tool that implements the [OCR-D
> CLI](https://ocr-d.de/en/spec/cli) and implements one or more
> [activities](#activities).

Reviewer: ```suggestion
A processor is a method provided by a tool that implements the uniform [OCR-D
CLI](https://ocr-d.de/en/spec/cli) for run-time data processing, and yields one or more
[activities](#activities).
```

→ [OCR-D Workflow Guide](https://ocr-d.de/en/workflows)

> ### Evaluator

Reviewer 2: this still needs to be translated to english

Author: 120f6a03

> CLI-Werkzeug welches die Ergebnis-Annotation eines bestimmten Workflow-Schrittes oder Prozessors qualitativ bewertet und relativ zu einem gegebenen Schwellwert vollständigen oder partiellen Erfolg signalisiert

Reviewer: ```suggestion
An evaluator is a tool that implements the uniform OCR-D CLI for run-time quality estimation, assessing an [activity's](#activities) annotation (i.e. a [processor's](#processor) output) with some quality metric to yield a score and applying a given threshold against it to signal full or partial success/failure.
```

Author: 1fd6cb85

> ### Modul

Reviewer: ```suggestion
### Module
```

Sofware package/repository providing one or more processors or evaluators, possibly encompassing additional areas of functionality (training, format conversion, creation of GT, visualization)

Modules can comprise multiple methods/activities that are called [*processors*](#processor)
for OCR-D. There were [eight MP](https://ocr-d.de/en/module-projects) in the
second phase of OCR-D (2018-2020).

### Messaging

Messaging service on the basis of Publish/Subscribe architecture (or similar) to coordinate network components, in particular for the distribution of tasks and load balancing, as well as signalling processor/evaluator results.

> ### (OCR-D-)Workflow

Reviewer: ```suggestion
### (OCR-D) Workflow
```

> Configuration of activities of processors/evaluators and their parameterization depending on success.

Reviewer: ```suggestion
Combination of [activities](#activities) via concrete [processors](#processor) and [evaluators](#evaluator) and their parameterization configured as a sequence or lattice, depending on their success or failure.
```
